### ISSUE TITLE
Value collection does not work with multi-path + transform

### ISSUE THREAD

**Issue by nathanmarz** [nathanmarz/specter#64]
Example:

```clojure
(transform (multi-path VAL) (fn [a b] b) :a)
;; => Exception thrown
```

The solution is to change variable-pathed-path and fixed-pathed-path to generate rich handlers rather than the lean ones they're currently generating.

**Comment by aengelberg** [comment]
Value collection also does not work with `if-path`:
```clojure
(transform [:a (if-path even? (putval 1) (putval 2))] + {:a 1})
;; => Exception thrown
```
I assume this is a similar issue.

**Comment by nathanmarz** [comment]
It's the same underlying bug.

**Comment by nathanmarz** [comment]
Value collection also has issues during selection:

```clojure
(use 'com.rpl.specter)
(use 'com.rpl.specter.macros)

(def mytree {"e1" {"e2" {"e1" {:template 1}
                        "e2" {:template 2}}}})

(declarepath KeyAccumWalker [k])
(providepath KeyAccumWalker (if-path must STAY [ALL (collect-one FIRST) LAST (params-reset KeyAccumWalker)]))
(select (KeyAccumWalker :template) mytree)
;; => [["e1" ["e2" ["e1" {:template 1}]]] ["e1" ["e2" ["e2" {:template 2}]]]]
```

In this example, the problem is that if-path works by doing a compiled-traverse* which does not compose with value collection appropriately.

**Comment by nathanmarz** [comment]
A solution to this needs if-path and multi-path to be able to manipulate `vals` directly and pass them directly into invocation of `compiled-traverse*` or `compiled-transform`

**Comment by nathanmarz** [comment]
Fixed for 0.12.0
